everyvoice/demo/app.py

```diff
@@ -91,16 +107,78 @@ def synthesize_audio(
         config=config,
         output_key=output_key,
         device=device,
-        global_step=1,
-        vocoder_global_step=1,  # dummy value since the vocoder step is not used
+        global_step=text_to_spec_model.config.training.max_steps,
+        vocoder_global_step=vocoder_model.config.training.max_steps,
         vocoder_model=vocoder_model,
         vocoder_config=vocoder_config,
     )
     # move to device because lightning accumulates predictions on cpu
     predictions[0][output_key] = predictions[0][output_key].to(device)
     wav, sr = wav_writer.synthesize_audio(predictions[0])
+    torchaudio.save(
+        wav_writer.get_filename(basename, speaker, language),
+        # the vocoder output includes padding so we have to remove that
+        wav[0],
+        sr,
+        format="wav",
+        encoding="PCM_S",
+        bits_per_sample=16,
+    )
+    wav_output = wav_writer.get_filename(basename, speaker, language)
+    file_writer = None
+    file_output = None
+    if output_format == SynthesizeOutputFormats.readalong_html.name:
```

Comment:
This and all the other callback constructor calls really ought to be able to be replaced by a single call to `get_synthesis_output_callbacks`, no?

I realize you need the wav writer to create the RAS_html writer, but that's already done too.

Refactoring suggestion: have `get_synthesis_output_callbacks` return a dict where the key are the output types, and the values are the writers. Then here you can use `writers["wav"]` and `writers[output_format]` to access the two writers you need, having passed `["wav", output_format]` (or the proper Enum form if need be) as the `output_type` argument to `get_synthesis_output_callbacks`.

Reply:
yea, this is a good idea. I would definitely be in favour of this refactor, which I think would clean up some of the acrobatics we're currently doing in order to get the wav writer separately.